iotdb/src/main/java/org/apache/iotdb/db/engine/memtable/AbstractMemTable.java

```diff
@@ -104,30 +137,49 @@ public boolean isEmpty() {
   @Override
   public ReadOnlyMemChunk query(String deviceId, String measurement, TSDataType dataType,
       Map<String, String> props) {
-    return new ReadOnlyMemChunk(dataType, getSeriesData(deviceId, measurement, dataType), props);
+    TimeValuePairSorter sorter;
+    if (!checkPath(deviceId, measurement)) {
+      return null;
+    } else {
+      long undeletedTime = findUndeletedTime(deviceId, measurement);
+      IWritableMemChunk memChunk = memTableMap.get(deviceId).get(measurement);
+      IWritableMemChunk chunkCopy = new WritableMemChunk(dataType, memChunk.getTVList().clone());
+      chunkCopy.setTimeOffset(undeletedTime);
+      sorter = chunkCopy;
+    }
+    return new ReadOnlyMemChunk(dataType, sorter, props);
   }
 
-  private TimeValuePairSorter getSeriesData(String deviceId, String measurement, TSDataType dataType) {
-    if (!checkPath(deviceId, measurement)) {
-      return new WritableMemChunk(dataType);
+
+  private long findUndeletedTime(String deviceId, String measurement) {
+    String path = deviceId + PATH_SEPARATOR + measurement;
+    long undeletedTime = Long.MIN_VALUE;
+    for (Modification modification : modifications) {
+      if (modification instanceof  Deletion) {
+        Deletion deletion = (Deletion) modification;
+        if (deletion.getPathString().equals(path) && deletion.getTimestamp() > undeletedTime) {
```

Comment:
how about remove concating deviceId and measurement. Instead, use `if(deletion.getDeviceId().equals() && deletion.getMeasurement().equals())`

Reply:
The full path is concatenated during construction, so this change does not make sense.